### ISSUE TITLE
nnstreamer in yolov8

### ISSUE THREAD

**Issue by NihalAbdulRazack** [nnstreamer/nnstreamer#4584]
gst-launch-1.0 v4l2src ! videoconvert ! videoscale !
video/x-raw,width=320,height=320,format=RGB,pixel-aspect-ratio=1/1,framerate=30/1 ! tee name=t
t. ! queue ! tensor_converter ! other/tensors,num_tensors=1,types=uint8,format=static,dimensions=3:320:320:1 !
tensor_transform mode=arithmetic option=typecast:float32,add:0.0,div:255.0 !
queue leaky=2 max-size-buffers=2 !
tensor_filter framework=tensorflow2-lite model=/home/rishad/NIhal/yolo/best_saved_model/best_float16.tflite custom=Delegate:XNNPACK,NumThreads:4 latency=1 !
other/tensors,num_tensors=1,types=float32,format=static,dimensions=2100:84:1 !
tensor_transform mode=transpose option=1:0:2:3 !
tensor_decoder mode=bounding_boxes option1=yolov8 option2=/home/rishad/NIhal/yolo/label.txt option3=0 option4=320:320 option5=320:320 !
video/x-raw,width=320,height=320,format=RGBA ! mix.sink_0
t. ! queue ! mix.sink_1
compositor name=mix sink_0::zorder=2 sink_1::zorder=1 ! videoconvert ! autovideosink
** Message: 17:13:40.792: accl = cpu

** (gst-launch-1.0:55221): WARNING **: 17:13:40.793: Input/output tensors should be memcpy-ed rather than explicitly assigning its ptr when XNNPACK Delegate is used.

** (gst-launch-1.0:55221): WARNING **: 17:13:40.793: This could cause performance degradation if sizes of input/output tensors are large
INFO: Created TensorFlow Lite XNNPACK delegate for CPU.

** (gst-launch-1.0:55221): CRITICAL **: 17:13:40.834: Yolov8 bounding-box decoder accepts float32 input tensors only. The given input tensor is: Num_Tensors = 1, Tensors = [{"", (null), (null)}].

** (gst-launch-1.0:55221): CRITICAL **: 17:13:40.834: Yolov8 bounding-box decoder accepts float32 input tensors only. The given input tensor is: Num_Tensors = 1, Tensors = [{"", (null), (null)}].
WARNING: erroneous pipeline: could not link queue1 to tensorfilter0
Can you please explain what is the issue here and how to solve it , the model here is custom model i trained using yolov8n and for label.txt it is in format
1 withouthelmet
2 withhelmet

**Comment by anyj0527** [comment]
I think your model `best_float16.tflite` has output shape of `2100:6:1` (`float[1, 6, 2100]`), meaning 2100 boxes and 6 additional info (4 for coordinates and 2 for class logit). May try removing the line `other/tensors,num_tensors=1,types=float32,format=static,dimensions=2100:84:1` or setting with proper numbers

**Comment by NihalAbdulRazack** [comment]
gst-launch-1.0 v4l2src ! videoconvert ! videoscale ! video/x-raw,width=320,height=320,format=RGB,pixel-aspect-ratio=1/1,framerate=30/1 ! tee name=t t. ! queue ! tensor_converter ! other/tensors,num_tensors=1,types=uint8,format=static,dimensions=3:320:320:1 ! tensor_transform mode=arithmetic option=typecast:float32,add:0.0,div:255.0 ! queue leaky=2 max-size-buffers=2 ! tensor_filter framework=tensorflow2-lite model=/home/rishad/NIhal/yolo/best_saved_model/best_float16.tflite custom=Delegate:XNNPACK,NumThreads:4 latency=1 ! capsfilter caps="other/tensors,num_tensors=1,types=float32,format=static,dimensions=2100:6:1" ! tensor_transform mode=transpose option=1:0:2:3 ! tensor_decoder mode=bounding_boxes option1=yolov8 option2=/home/rishad/NIhal/yolo/label.txt option3=0 option4=320:320 option5=320:320 ! video/x-raw,width=320,height=320,format=RGBA ! mix.sink_0 t. ! queue ! mix.sink_1 compositor name=mix sink_0::zorder=2 sink_1::zorder=1 ! videoconvert ! autovideosink



** Message: 12:01:26.341: accl = cpu

** (gst-launch-1.0:8803): WARNING **: 12:01:26.341: Input/output tensors should be memcpy-ed rather than explicitly assigning its ptr when XNNPACK Delegate is used.

** (gst-launch-1.0:8803): WARNING **: 12:01:26.341: This could cause performance degradation if sizes of input/output tensors are large
INFO: Created TensorFlow Lite XNNPACK delegate for CPU.
WARNING: erroneous pipeline: could not link queue1 to tensorfilter0.
Now the output shape is resolved but the problem now is the linking of queue.Sorry iam new to gstreamer, Thanks for the help

**Comment by anyj0527** [comment]
Make clear that your model has shape of 2100:6:1 first. The pipeline should work without any `capsfilter` or `other/tensors,...`.
Try to make your pipeline work without it. You may find it easy if you make small pipeline (without tensor_decoder) work first, then attach downstream elements.

**Comment by NihalAbdulRazack** [comment]
Input 0:
Shape: [  1 640 640   3]
Data Type: <class 'numpy.float32'>

Output details:
Output 0:
Shape: [   1    6 8400]
Data Type: <class 'numpy.float32'>
_**This is the output shape that i get by running a code that use interpreter**_
 
gst-launch-1.0 v4l2src ! videoconvert ! videoscale ! video/x-raw,width=320,height=320,format=RGB,pixel-aspect-ratio=1/1,framerate=30/1 ! tee name=t t. ! queue ! tensor_converter ! other/tensors,num_tensors=1,types=uint8,format=static,dimensions=3:320:320:1 ! tensor_transform mode=arithmetic option=typecast:float32,add:0.0,div:255.0 ! queue leaky=2 max-size-buffers=2 ! tensor_filter framework=tensorflow2-lite model=/home/rishad/NIhal/yolo/best_saved_model/best_float16.tflite custom=Delegate:XNNPACK,NumThreads:4 latency=1 ! capsfilter caps="other/tensors,num_tensors=1,types=float32,format=static,dimensions=8400:6:1" ! tensor_transform mode=transpose option=1:0:2:3 ! tensor_decoder mode=bounding_boxes option1=yolov8 option2=/home/rishad/NIhal/yolo/label.txt option3=0 option4=320:320 option5=320:320 ! video/x-raw,width=320,height=320,format=RGBA ! mix.sink_0 t. ! queue ! mix.sink_1 compositor name=mix sink_0::zorder=2 sink_1::zorder=1 ! videoconvert ! autovideosink
** Message: 10:17:39.124: accl = cpu

** (gst-launch-1.0:14204): WARNING **: 10:17:39.125: Input/output tensors should be memcpy-ed rather than explicitly assigning its ptr when XNNPACK Delegate is used.

** (gst-launch-1.0:14204): WARNING **: 10:17:39.125: This could cause performance degradation if sizes of input/output tensors are large
INFO: Created TensorFlow Lite XNNPACK delegate for CPU.

** (gst-launch-1.0:14204): CRITICAL **: 10:17:39.167: yolov8 boundingbox decoder requires the input shape to be 6:2100:1. But given shape is 6:8400:1. `tensor_transform mode=transpose` would be helpful.

** (gst-launch-1.0:14204): CRITICAL **: 10:17:39.168: yolov8 boundingbox decoder requires the input shape to be 6:2100:1. But given shape is 6:8400:1. `tensor_transform mode=transpose` would be helpful.
WARNING: erroneous pipeline: could not link queue1 to tensorfilter0
_**But when i use this shape it is telling that input shape needed to 6:2100:1**_

gst-launch-1.0 v4l2src ! videoconvert ! videoscale ! video/x-raw,width=320,height=320,format=RGB,pixel-aspect-ratio=1/1,framerate=30/1 ! tee name=t t. ! queue ! tensor_converter ! other/tensors,num_tensors=1,types=uint8,format=static,dimensions=3:320:320:1 ! tensor_transform mode=arithmetic option=typecast:float32,add:0.0,div:255.0 ! queue leaky=2 max-size-buffers=2 ! tensor_filter framework=tensorflow2-lite model=/home/rishad/NIhal/yolo/best_saved_model/best_float16.tflite custom=Delegate:XNNPACK,NumThreads:4 latency=1 ! tensor_transform mode=transpose option=1:0:2:3 ! tensor_decoder mode=bounding_boxes option1=yolov8 option2=/home/rishad/NIhal/yolo/label.txt option3=0 option4=320:320 option5=320:320 ! video/x-raw,width=320,height=320,format=RGBA ! mix.sink_0 t. ! queue ! mix.sink_1 compositor name=mix sink_0::zorder=2 sink_1::zorder=1 ! videoconvert ! autovideosink
** Message: 10:21:07.132: accl = cpu

** (gst-launch-1.0:14298): WARNING **: 10:21:07.133: Input/output tensors should be memcpy-ed rather than explicitly assigning its ptr when XNNPACK Delegate is used.

** (gst-launch-1.0:14298): WARNING **: 10:21:07.133: This could cause performance degradation if sizes of input/output tensors are large
INFO: Created TensorFlow Lite XNNPACK delegate for CPU.

** (gst-launch-1.0:14298): CRITICAL **: 10:21:07.177: int check_tensors(const GstTensorsConfig*, unsigned int): assertion 'config->info.num_tensors >= limit' failed

** (gst-launch-1.0:14298): CRITICAL **: 10:21:07.177: Yolov8 bounding-box decoder needs at least 1 valid tensor. The given input tensor is: Num_Tensors = 0, Tensors = [].

** (gst-launch-1.0:14298): CRITICAL **: 10:21:07.177: int check_tensors(const GstTensorsConfig*, unsigned int): assertion 'config->info.num_tensors >= limit' failed

** (gst-launch-1.0:14298): CRITICAL **: 10:21:07.177: Yolov8 bounding-box decoder needs at least 1 valid tensor. The given input tensor is: Num_Tensors = 0, Tensors = [].
WARNING: erroneous pipeline: could not link queue1 to tensorfilter0

**_THIS IS WHAT I GET WHEN I REMOVE THE CAPSFILTER_**

**Comment by NihalAbdulRazack** [comment]
Actually i solved this issue , i changed the input shape to  [ 1 640 640 3] and output shape was 8400:6:1, this resolved the issue , Thank you for the help, your comment made me focus on shape issue.
